# Description: Ensure that clipboard_aura.o is not omitted from the build
# Origin: http://src.chromium.org/viewvc/chrome?view=revision&revision=241390

diff --git a/build/common.gypi b/build/common.gypi
--- a/build/common.gypi
+++ b/build/common.gypi
@@ -547,20 +547,16 @@
 
         # Flags to use X11 on non-Mac POSIX platforms.
         ['OS=="win" or OS=="mac" or OS=="ios" or OS=="android" or use_ozone==1', {
           'use_x11%': 0,
         }, {
           'use_x11%': 1,
         }],
 
-        ['OS=="linux" and use_aura==1 and chromeos==0', {
-          'use_clipboard_aurax11%': 1,
-        }],
-
         # Flags to use glib.
         ['OS=="win" or OS=="mac" or OS=="ios" or OS=="android" or embedded==1', {
           'use_glib%': 0,
         }, {
           'use_glib%': 1,
         }],
 
         # Flags to use pango and cairo.
@@ -1901,16 +1897,21 @@
 
         # Enable built-in ozone platforms if ozone is enabled.
         'ozone_platform_dri%': 1,
         'ozone_platform_test%': 1,
       }, {  # use_ozone==0
         'ozone_platform_dri%': 0,
         'ozone_platform_test%': 0,
       }],
+
+      ['desktop_linux==1 and use_aura==1 and use_x11==1', {
+        'use_clipboard_aurax11%': 1,
+      }],
+
       ['OS=="win" and use_goma==1', {
         # goma doesn't support pch yet.
         'chromium_win_pch': 0,
         # goma doesn't support PDB yet, so win_z7=1 or fastbuild=1.
         'conditions': [
           ['fastbuild==0', {
             'win_z7': 1,
           }],
diff --git a/ui/ui.gyp b/ui/ui.gyp
--- a/ui/ui.gyp
+++ b/ui/ui.gyp
@@ -420,17 +420,17 @@
             }],
           ],
         }],
         ['use_pango==1', {
           'dependencies': [
             '../build/linux/system.gyp:pangocairo',
           ],
         }],
-        ['use_x11==0 or use_clipboard_aurax11==1', {
+        ['OS=="win" or use_clipboard_aurax11==1', {
           'sources!': [
             'base/clipboard/clipboard_aura.cc',
           ],
         }, {
           'sources!': [
             'base/clipboard/clipboard_aurax11.cc',
           ],
         }],
